extensions/amp-story-interactive/1.0/amp-story-interactive-abstract.js

```diff
@@ -233,29 +248,37 @@ export class AmpStoryInteractive extends AMP.BaseElement {
     return this.pageId_;
   }
 
-  /**
-   * Initializes the component when the amp-story is created.
-   * @public
-   */
-  initializeState() {
-    this.updateStoryStoreState_(null);
-  }
-
   /** @override */
   buildCallback(concreteCSS = '') {
     this.loadFonts_();
     this.options_ = this.parseOptions_();
-    this.rootEl_ = this.buildComponent();
-    this.rootEl_.classList.add('i-amphtml-story-interactive-container');
-    this.element.classList.add('i-amphtml-story-interactive-component');
     this.adjustGridLayer_();
-    this.initializeListeners_();
     devAssert(this.element.children.length == 0, 'Too many children');
-    createShadowRootWithStyle(
-      this.element,
-      dev().assertElement(this.rootEl_),
-      CSS + concreteCSS
-    );
+    return Promise.all([
+      Services.storyVariableServiceForOrNull(this.win).then((service) => {
+        this.variableService_ = service;
+      }),
+      Services.storyStoreServiceForOrNull(this.win).then((service) => {
+        this.storeService_ = service;
+        this.updateStoryStoreState_(null);
+      }),
+      Services.storyRequestServiceForOrNull(this.win).then((service) => {
+        this.requestService_ = service;
+      }),
+      Services.storyAnalyticsServiceForOrNull(this.win).then((service) => {
+        this.analyticsService_ = service;
+      }),
+    ]).then(() => {
+      this.rootEl_ = this.buildComponent();
+      this.rootEl_.classList.add('i-amphtml-story-interactive-container');
+      this.element.classList.add('i-amphtml-story-interactive-component');
+      this.initializeListeners_();
+      createShadowRootWithStyle(
+        this.element,
+        dev().assertElement(this.rootEl_),
+        CSS + concreteCSS
+      );
```

Comment:
This should be done in the `buildCallback`, not after as it's working rn.
You could do all of it before your `Promise.all` statement, and move the `initializeListeners_()` to `layoutCallback()`.